src/gapc.cc

```diff
@@ -602,6 +634,37 @@ class Main {
         + opts.plot_grammar_file + " > foo.pdf' to generate a PDF.");
     }
 
+    if (opts.checkpoint_interval > 0) {
+      // list of currently supported/serializable datatypes (all primitive)
+      constexpr Type::Type
+      SUPPORTED_TYPES[] = {Type::Type::VOID, Type::Type::INTEGER,
+                           Type::Type::INT, Type::Type::FLOAT,
+                           Type::Type::SIZE, Type::Type::SINGLE,
+                           Type::Type::BIGINT};
+
+      // answer type
+      Type::Type answer_type = driver.ast.grammar()->axiom->
```

Comment:
In principle, a signature (or grammar) can have multiple different answer types, see e.g. https://github.com/jlab/fold-grammars/blob/1a7cf6e861c1af398edc4f1db8794792e7f99cd2/Signatures/sig_pknot_foldrna.gap#L1
Furthermore, when NOT applying the choice function to a set of alternative production rules, the answer is an explicit "list of ", while often the compiler simplifies the return type to e.g. a float/double: play with https://github.com/jlab/fold-grammars/blob/1a7cf6e861c1af398edc4f1db8794792e7f99cd2/Grammars/Parts/grapart_basic.gap#L3 and remove the `# h` application for the "axiom" production rule.